desc/objectives/_neoclassical.py

```diff
@@ -0,0 +1,263 @@
+    wells that lead to bad confinement properties with enhanced radial drift,
+    especially for trapped particles. Neoclassical (thermal) transport can become the
+    dominant transport channel in stellarators which are not optimized to reduce it.
+    The effective ripple is a proxy, measuring the effective modulation amplitude of the
+    magnetic field averaged along a magnetic surface, which can be used to optimize for
+    stellarators with improved confinement. It is targeted as a flux surface function.
+
+    References
+    ----------
+    https://doi.org/10.1063/1.873749.
+    Evaluation of 1/ν neoclassical transport in stellarators.
+    V. V. Nemov, S. V. Kasilov, W. Kernbichler, M. F. Heyn.
+    Phys. Plasmas 1 December 1999; 6 (12): 4622–4632.
+
+    Parameters
+    ----------
+    eq : Equilibrium
+        Equilibrium that will be optimized to satisfy the Objective.
+    target : {float, ndarray, callable}, optional
+        Target value(s) of the objective. Only used if bounds is None.
+        Must be broadcastable to Objective.dim_f. If a callable, should take a
+        single argument ``rho`` and return the desired value of the profile at those
+        locations. Defaults to 0.
+    bounds : tuple of {float, ndarray, callable}, optional
+        Lower and upper bounds on the objective. Overrides target.
+        Both bounds must be broadcastable to Objective.dim_f.
+        If a callable, each should take a single argument ``rho`` and return the
+        desired bound (lower or upper) of the profile at those locations.
+    weight : {float, ndarray}, optional
+        Weighting to apply to the Objective, relative to other Objectives.
+        Must be broadcastable to Objective.dim_f
+    normalize : bool, optional
+        Whether to compute the error in physical units or non-dimensionalize.
+    normalize_target : bool, optional
+        Whether target and bounds should be normalized before comparing to computed
+        values. If `normalize` is ``True`` and the target is in physical units,
+        this should also be set to True.
+    loss_function : {None, 'mean', 'min', 'max'}, optional
+        Loss function to apply to the objective values once computed. This loss function
+        is called on the raw compute value, before any shifting, scaling, or
+        normalization.
+    deriv_mode : {"auto", "fwd", "rev"}
+        Specify how to compute Jacobian matrix, either forward mode or reverse mode AD.
+        "auto" selects forward or reverse mode based on the size of the input and output
+        of the objective. Has no effect on self.grad or self.hess which always use
+        reverse mode and forward over reverse mode respectively.
+    rho : ndarray
+        Unique coordinate values specifying flux surfaces to compute on.
+    alpha : ndarray
+        Unique coordinate values specifying field line labels to compute on.
+    knots_per_transit : int
```

Comment:
will these be removed after #1119 ?

Reply:
No this will same as the `N` chebyshev resolution used to construct $\theta$.

Reply:
in that case do we need to rename it? Is the plan to have a single objective for eps_eff? or two different ones for 1d/2d? I'd prefer the former

Reply:
#1290 replaces the objectives here.